adserver/analyzer/models.py

```diff
@@ -0,0 +1,62 @@
+"""Stored results of offline content targeting analysis."""
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+from django_extensions.db.models import TimeStampedModel
+from jsonfield import JSONField
+from simple_history.models import HistoricalRecords
+
+from ..models import Publisher
+
+
+class AnalyzedUrl(TimeStampedModel):
+
+    """Analyzed keywords for a given URL."""
+
+    # Query parameters ignored by the analyzer
+    IGNORE_QUERY_PARAMS = (
+        "q",
+        "query",
+        "search",
+    )
+
+    url = models.URLField(
+        db_index=True,
+        max_length=1024,
+        help_text=_(
+            "URL of the page being analyzed after certain query parameters are stripped away"
+        ),
+    )
+
+    publisher = models.ForeignKey(
+        Publisher,
+        help_text=_("Publisher where this URL appears"),
+        on_delete=models.CASCADE,
+    )
+
+    # Fields below are updated by the analyzer
+    keywords = JSONField(_("Keywords for this URL"), blank=True, null=True)
```

Comment:
Is there a good way to document the presumed schema for this JSON? Is the model the right place for it?

Reply:
I was planning to add a validator.